library/src/main/java/com/dslplatform/json/processor/Analysis.java

```diff
@@ -697,11 +697,38 @@ public void findRelatedReferences() {
 						}
 					}
 				}
+				findGenericAttributes(info);
 				path.pop();
 			}
 		} while (total != structs.size());
 	}
 
+	private void findGenericAttributes(StructInfo si) {

```

Comment:
This looks like it can be done when we register the attribute, instead of later changing the attribute with genericType info

Reply:
Might be better if we avoid the genericType and fix the type (?) attribute which should point to the correct class. The problem is the concrete class may be lost if we go deep into parent. This method in general is a workaround, for example `declaredType.getTypeArguments()` return correct class. If we do `Element element = declaredType.asElement()` the element will lost the information about this class and return the parameter value `T`.

Reply:
@varahash added the support for generics so he might know better
I know of two main cases:
 * where we need runtime info to fully construct the correct type (in which case we pass in some additional type info to the converter and do it in runtime)
 * where we have all the info at compile time

Here is the conversion for the runtime case: https://github.com/ngs-doo/dsl-json/blob/master/java8/src/main/java/com/dslplatform/json/processor/ConverterTemplate.java#L274
But compile time case should be just a type?